src/submap.h

```diff
@@ -288,8 +289,9 @@ struct maptile {
             return sm->itm[x][y].size();
         }
 
+        // Assumes there is at least one item
         const item &get_uppermost_item() const {
-            return sm->itm[x][y].back();
+            return *std::prev( sm->itm[x][y].cend() );
```

Comment:
Yeah that's exactly what it's for.

And yeah that's for a future PR.  I've also considered rewriting some of the inventory code to display alphabetically since storage on the map is no longer ordered.